v1.go

```diff
@@ -22,17 +23,20 @@ func initVersion1() *Header {
 }
 
 func parseVersion1(reader *bufio.Reader) (*Header, error) {
-	// Read until LF shows up, otherwise fail.
-	// At this point, can't be sure CR precedes LF which will be validated next.
-	line, err := reader.ReadString('\n')
-	if err != nil {
+	// The header must be terminated by CRLF and cannot be more than 107 bytes long.
+	buf, err := reader.Peek(107)
+	if err != nil && err != io.EOF {
 		return nil, ErrLineMustEndWithCrlf
 	}
-	if !strings.HasSuffix(line, crlf) {
+	pos := bytes.Index(buf, []byte(crlf))
+	if pos == -1 {
```

Comment:
The existing tests already capture this.

Reply:
Not clear what you mean by this. The `Peek` can return a buffer without CRLF at this point. It can also return a buffer that extends beyond the CRLF (e.g. short header, but more bytes available in the stream). So `HasSuffix` does not work here. Perhaps I've misunderstood your point though...

Reply:
Just a comment that existing tests (in main branch) already capture a scenario where the string read has LF but not CRLF.